### PR TITLE
Refactor reference implementation from lite/kernels/internal/reference/reference_ops.h into own headers

### PR DESCRIPTION
**Pull request by stephanboner**
PR2 from #45693

### CODE REVIEW

**Review by thaink — APPROVED** [review]

File: tensorflow/lite/kernels/internal/reference/batch_to_space_nd.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_BATCH_TO_SPACE_ND_H_
+#define TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_BATCH_TO_SPACE_ND_H_
+
+#include <cmath>
+
+#include "tensorflow/lite/kernels/internal/types.h"
+
```
Comment (thaink) [inline tensorflow/lite/kernels/internal/reference/batch_to_space_nd.h:21]: #include "ruy/profiler/instrumentation.h"  // from @ruy
Can you above add include? our internal tests are failling.

File: tensorflow/lite/kernels/internal/reference/space_to_batch_nd.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_SPACE_TO_BATCH_ND_H_
+#define TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_SPACE_TO_BATCH_ND_H_
+
+#include <cmath>
+
+#include "tensorflow/lite/kernels/internal/types.h"
+
```
Comment (thaink) [inline tensorflow/lite/kernels/internal/reference/space_to_batch_nd.h:21]: ditto

File: tensorflow/lite/kernels/internal/reference/batch_to_space_nd.h (hunk 1 above)
Comment (stephanboner) [reply tensorflow/lite/kernels/internal/reference/batch_to_space_nd.h:21]: Thanks, done!

File: tensorflow/lite/kernels/internal/reference/space_to_batch_nd.h (hunk 2 above)
Comment (stephanboner) [reply tensorflow/lite/kernels/internal/reference/space_to_batch_nd.h:22]: as well :)

**Review by thaink — APPROVED** [review]
Re-approved the change.

**Comment by gbaned** [comment]
@stephanboner Can you please resolve conflicts? Thanks!

**Comment by stephanboner** [comment]
@gbaned done!

**Comment by thaink** [comment]
This PR is being review internally due to a test failure. I am waiting for an approval before merging it.

**Comment by thaink** [comment]
This one is merged in https://github.com/tensorflow/tensorflow/commit/3654281b029c3186bb2168ca20a4e1558f0620eb

**Pull request closed without merging**